tests/test_minimal/test_tools/aws_tools.py

```diff
@@ -0,0 +1,158 @@
+    info = submit_aws_batch_job(job_name=job_name, docker_image=docker_image, commands=commands)
+
+    # Wait for AWS to process the job
+    time.sleep(60)
+
+    job_id = info["job_submission_info"]["jobId"]
+
+    all_jobs_response = batch_client.describe_jobs(jobs=[job_id])
+    assert all_jobs_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    jobs = all_jobs_response["jobs"]
+    assert len(jobs) == 1
+
+    job = jobs[0]
+    assert job["jobName"] == job_name
+    assert "neuroconv_batch_queue" in job["jobQueue"]
+    assert "neuroconv_batch_ubuntu-latest-image_4-GiB-RAM_4-CPU" in job["jobDefinition"]
+    assert job["status"] == "SUCCEEDED"
+
+    status_tracker_table_name = "neuroconv_batch_status_tracker"
+    table = dynamodb_resource.Table(name=status_tracker_table_name)
+    table_submission_id = info["table_submission_info"]["id"]
+
+    table_item_response = table.get_item(Key={"id": table_submission_id})
+    assert table_item_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    table_item = table_item_response["Item"]
+    assert table_item["job_name"] == job_name
+    assert table_item["job_id"] == job_id
+    assert table_item["status"] == "Job submitted..."
+
+    table.update_item(
+        Key={"id": table_submission_id}, AttributeUpdates={"status": {"Action": "PUT", "Value": "Test passed."}}
+    )
+
+
+def test_submit_aws_batch_job_with_dependencies():
+    region = "us-east-2"
+    aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID", None)
+    aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY", None)
+
+    dynamodb_resource = boto3.resource(
+        service_name="dynamodb",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+    batch_client = boto3.client(
+        service_name="batch",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+
+    job_name_1 = "test_submit_aws_batch_job_with_dependencies_1"
+    docker_image = "ubuntu:latest"
+    commands_1 = ["echo", "'Testing NeuroConv AWS Batch submission.'"]
+
+    # TODO: to reduce costs even more, find a good combinations of memory/CPU to minimize size of instance
+    job_info_1 = submit_aws_batch_job(
+        job_name=job_name_1,
+        docker_image=docker_image,
+        commands=commands_1,
+    )
+    job_submission_info_1 = job_info_1["job_submission_info"]
+
+    job_name_2 = "test_submit_aws_batch_job_with_dependencies_1"
+    commands_2 = ["echo", "'Testing NeuroConv AWS Batch submission with dependencies.'"]
+    job_dependencies = [{"jobId": job_submission_info_1["jobId"], "type": "SEQUENTIAL"}]
+    job_info_2 = submit_aws_batch_job(
+        job_name=job_name_2,
+        docker_image=docker_image,
+        commands=commands_2,
+        job_dependencies=job_dependencies,
+    )
+
+    # Wait for AWS to process the jobs
+    time.sleep(120)
```

Comment:
Isn't this brittle? Can we query directly to see if the job is concluded?

Reply:
Technically yes

If you want something better we could do the same retry mechanism as seen in source code

But this is such a simple job and has not had trouble so far just waiting a static amount of time

Whichever you prefer